lib/salesforce_sync/resource/action.rb

```diff
@@ -1,16 +1,39 @@
 module SalesforceSync
   module Resource
     class Action
+      SF_USER_QUERY_TYPE = "Contact"
+
       def initialize(sf_class = nil, resource_id = nil)
         @sf_class = sf_class
         @resource_id = resource_id
       end
 
       def upsert
-        if sf_resource.resource.present?
-          salesforce_id = client.upsert!(sf_class.sf_type, sf_class.resource_id_name, sf_resource.all_prepared_fields)
-          sf_resource.store_salesforce_id(salesforce_id)
-          sf_resource.after_upsert
+        raise_missing_record unless sf_resource.resource.present?
+
+        if sf_resource.synchronised?
+          # its already synchronised so we can upsert
```

Comment:
Left these comments in to help decipher the logic for the PR - can remove before deploying